src/types/index.d.ts

```diff
@@ -1176,6 +1176,22 @@ export interface CoreScaleOptions {
    * @default true
    */
   weight: number;
+  /**
+   * User defined minimum value for the scale, overrides minimum value from data.
+   */
+  min: unknown;
+  /**
+   * User defined maximum value for the scale, overrides maximum value from data.
+   */
+  max: unknown;
+  /**
+   * Adjustment used when calculating the maximum data value.
+   */
+  suggestedMin: unknown;
+  /**
+   * Adjustment used when calculating the minimum data value.
+   */
+  suggestedMax: unknown;
```

Comment:
Thought this does not work with extending the scale since the props are defined here it does not take them from the specifics anymore so they will have to be ommited from every other override if I remember correctly.

Same got done in the `CategoryScaleOptions`

Reply:
@LeeLenaleee, I put together a quick demo in [TypeScript Playground](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgMIHsoQMoLgGwgHkAHMYdEAZ2QG8AoZJ5AW1AC5kBXEAaxHQB3EAG5GzFnAAenHvyGj6AX3r1QkWIhSo4USFWBwQuAsTIVqyCFMggAJjQxYThUuUo0GzVh2QguLABG0GLekjJ+AcFQYir0MDwI7iDIABZGdoQAagRcEAAUAG65EJz+QdAAlHTiTAge6IQAdPjoAOZFJZWxqvXUYMj1WJxOOHiu5h7IALx0PiCcAORwiwA0rNJLgYvISmJDEE1sKbMAjGLp9tkl+QdHoN3IAPRPVlIkEEkQdlZQUJjIAC0aEwY1MbgsVHuKWANDkAmEvQ8AzweggBiMI10+kMxnGZmSNFmtHmnFO63CnAATLt9tj0bjoTNkOd6JdMhAcvg8rd6RiQNDuvRUTijEzZstFiImC83h8vj9oP8oEC0HzcS4CZCmSwuFQBsFkHBIhUoEA), and I think everything is working as expected and desired.

However, I know that Chart.js is a widely used project, and its type definitions have some complexity to them, so if I'm missing something, please let me know.